plugins/obs-filters/cmake/rnnoise.cmake

```diff
@@ -1,47 +1,52 @@
 option(ENABLE_RNNOISE "Enable building with RNNoise noise supression filter" ON)
 
 if(ENABLE_RNNOISE)
-  if(NOT OS_WINDOWS AND NOT OS_MACOS)
-    add_library(obs-rnnoise OBJECT)
-
-    target_sources(
-      obs-rnnoise
-      PRIVATE # cmake-format: sortable
-              rnnoise/src/_kiss_fft_guts.h
-              rnnoise/src/arch.h
-              rnnoise/src/celt_lpc.c
-              rnnoise/src/celt_lpc.h
-              rnnoise/src/common.h
-              rnnoise/src/denoise.c
-              rnnoise/src/kiss_fft.c
-              rnnoise/src/kiss_fft.h
-              rnnoise/src/opus_types.h
-              rnnoise/src/pitch.c
-              rnnoise/src/pitch.h
-              rnnoise/src/rnn.c
-              rnnoise/src/rnn.h
-              rnnoise/src/rnn_data.c
-              rnnoise/src/rnn_data.h
-              rnnoise/src/rnn_reader.c
-              rnnoise/src/tansig_table.h
-      PUBLIC rnnoise/include/rnnoise.h)
-
-    add_library(Librnnoise::Librnnoise ALIAS obs-rnnoise)
-
-    target_include_directories(obs-rnnoise PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/rnnoise/include")
-
-    target_compile_definitions(obs-rnnoise PUBLIC COMPILE_OPUS)
-
-    target_compile_options(obs-rnnoise PRIVATE -Wno-newline-eof -Wno-error=null-dereference)
-
-    set_target_properties(obs-rnnoise PROPERTIES FOLDER plugins/obs-filters/rnnoise POSITION_INDEPENDENT_CODE TRUE)
-  else()
+  if(OS_WINDOWS OR OS_MACOS)
     find_package(Librnnoise REQUIRED)
-
-    target_sources(obs-filters PRIVATE noise-suppress-filter.c)
+  else()
+    find_package(Librnnoise)
+
+    if(NOT TARGET Librnnoise::Librnnoise)
+      add_library(obs-rnnoise OBJECT)
```

Comment:
We should maybe provide a message that we're using the in-tree RNNoise because a failed `find_package` call will emit a confusing warning otherwise:
```
CMake Warning (dev) at cmake/finders/FindLibrnnoise.cmake:89 (message):
  Failed to find Librnnoise version.
Call Stack (most recent call first):
  plugins/obs-filters/cmake/rnnoise.cmake:7 (find_package)
  plugins/obs-filters/CMakeLists.txt:36 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find Librnnoise (missing: Librnnoise_LIBRARY Librnnoise_INCLUDE_DIR) (found version "0.0.0")
    Reason given by package: Ensure librnnoise libraries are available in local libary paths.
```

Reply:
Bump on this.

Reply:
Seems fixed in latest push. Resolving thread.